src/views/CashlinkManage.vue

```diff
@@ -0,0 +1,342 @@
+    private retrievedCashlink: Cashlink | null = null;
+    private copied = false;
+    private nativeShareAvailable: boolean = false;
+
+    private sharePrefix = 'Here is your Nimiq Cashlink!';
+
+    private async mounted() {
+        // @ts-ignore Property 'share' does not exist on type 'Navigator'
+        this.nativeShareAvailable = (!!window.navigator && !!window.navigator.share);
+        this.retrievedCashlink = this.cashlink;
+        if (!this.keyguardResult) {
+            // If there is no Keyguard result this is not a freshly funded cashlink
+            // and must be retrieved from the store.
+            if (this.request.cashlinkAddress) {
+                const cashlink = await CashlinkStore.Instance.get(this.request.cashlinkAddress.toUserFriendlyAddress());
+                if (cashlink) {
+                    this.retrievedCashlink = cashlink;
+                    this.isTxSent = true;
+                    this.isManagementRequest = true;
+                }
+            }
+        }
+
+        if (!this.isTxSent) {
+            if (!NetworkClient.hasInstance()) {
+                NetworkClient.createInstance(Config.networkEndpoint);
+            }
+            const network = NetworkClient.Instance;
+            await network.init();
+
+            if (!this.keyguardResult || !this.keyguardRequest) {
+                this.$rpc.reject(new Error('Unexpected: No valid Cashlink;'));
+                return;
+            }
+            // if there was a funding transaction the cashlink is in the static store.
+            this.retrievedCashlink.networkClient = network;
+            network.on(NetworkClient.Events.API_READY,
+                () => this.status = 'Contacting seed nodes...');
+            network.on(NetworkClient.Events.CONSENSUS_SYNCING,
+                () => this.status = 'Syncing consensus...');
+            network.on(NetworkClient.Events.CONSENSUS_ESTABLISHED,
+                () => this.status = 'Sending transaction...');
+            network.on(NetworkClient.Events.TRANSACTION_PENDING,
+                () => this.status = 'Awaiting receipt confirmation...');
+
+            network.relayTransaction({
+                sender: new Nimiq.Address(this.keyguardRequest.sender).toUserFriendlyAddress(),
+                senderPubKey: this.keyguardResult.publicKey,
+                recipient:  new Nimiq.Address(this.keyguardRequest.recipient).toUserFriendlyAddress(),
+                value: Nimiq.Policy.lunasToCoins(this.keyguardRequest.value),
+                fee: Nimiq.Policy.lunasToCoins(this.keyguardRequest.fee),
+                validityStartHeight: this.keyguardRequest.validityStartHeight,
+                signature: this.keyguardResult.signature,
+                extraData: this.keyguardRequest.data,
+            });
+
+            network.on(NetworkClient.Events.TRANSACTION_RELAYED, async (txInfo: any) => {
+                await CashlinkStore.Instance.put(this.retrievedCashlink!);
```

Comment:
Maybe add error handling around this.

Reply:
What would be the expected Error here?
We have not done any Error handling on the `Store.put(...)` anywhere.

Reply:
Hum, interesting.
Shouldn't we generally handle errors in writing stuff to the indexed db?

Reply:
I agree. As this is already too big now I suggest doing it later in another PR.

Reply:
I opened an issue for this: https://github.com/nimiq/hub/issues/363